tests/test_helpers.py

```diff
@@ -0,0 +1,131 @@
+
+
+class NavigationContentTypeSearchTestCase(CMSTestCase):
+
+    def setUp(self):
+        self.language = settings.LANGUAGES[0][0]
+
+    def test_page_content_type_in_node_tree(self):
+        """
+        Test to ensure the correct node mapped to poll content is returned by the helper"
+        """
+        menu_contents = factories.MenuContentFactory()
+        factories.ChildMenuItemFactory(parent=menu_contents.root)
+        page_content = factories.PageContentWithVersionFactory(
+            language=self.language, version__created_by=self.get_superuser()
+        )
+        child2 = factories.ChildMenuItemFactory(parent=menu_contents.root, content=page_content)
+
+        result = get_navigation_node_for_content_object(menu_contents, page_content)
+
+        self.assertEqual(result, child2)
+
+    def test_to_search_content_object_in_nested_node_tree_search(self):
+        """
+        Test to ensure  the first node is returned by the helper if more than one node is mapped to content object
+        """
+        menu_contents = factories.MenuContentFactory()
+        child1 = factories.ChildMenuItemFactory(parent=menu_contents.root)
+        page_content = factories.PageContentWithVersionFactory(
+            language=self.language, version__created_by=self.get_superuser()
+        )
+        factories.ChildMenuItemFactory(parent=menu_contents.root)
+        grandchild = factories.ChildMenuItemFactory(parent=child1)
+        grandchild1 = factories.ChildMenuItemFactory(parent=grandchild)
+        grandchild2 = factories.ChildMenuItemFactory(parent=grandchild1, content=page_content)
+
+        result = get_navigation_node_for_content_object(menu_contents, page_content)
+
+        self.assertEqual(result, grandchild2)
+
+    def test_content_object_not_found_in_node_tree(self):
+        """
+        Test to ensure False if returned  by the helper when no node is mapped to content in node tree
+        """
+        menu_contents = factories.MenuContentFactory()
+        child1 = factories.ChildMenuItemFactory(parent=menu_contents.root)
+        page_content = factories.PageContentWithVersionFactory(
+            language=self.language, version__created_by=self.get_superuser()
+        )
+        factories.ChildMenuItemFactory(parent=menu_contents.root)
+
+        result = get_navigation_node_for_content_object(menu_contents, page_content)
+
+        self.assertFalse(result)
+
+        grandchild = factories.ChildMenuItemFactory(parent=child1)
+        grandchild1 = factories.ChildMenuItemFactory(parent=grandchild)
+        factories.ChildMenuItemFactory(parent=grandchild1)
+
+        result = get_navigation_node_for_content_object(menu_contents, page_content)
+
+        self.assertFalse(result)
+
+    def test_content_object_mapped_to_more_than_node(self):
+        """
+        Test to ensure first node found returned by helper when more than one node is mapped to content object
```

Comment:
The first node found returned is returned by the helper when more than one node is mapped to a content object in a node tree